dbms/src/Storages/transformQueryForExternalDatabase.cpp

```diff
@@ -13,17 +14,57 @@
 namespace DB
 {
 
+Block getBlockWithConstants(
+    const ASTPtr & query, const Context & context, const NamesAndTypesList & all_columns)
+{
+    Block result
+    {
+        { DataTypeUInt8().createColumnConstWithDefaultValue(1), std::make_shared<DataTypeUInt8>(), "_dummy" }
+    };
+
+    const auto expr_for_constant_folding = ExpressionAnalyzer{query, context, nullptr, all_columns}.getConstActions();
+
+    expr_for_constant_folding->execute(result);
+
+    return result;
+}
+
+static void replaceConstFunction(const IAST & node, const Context & context, const NamesAndTypesList & all_columns)
+{
+    for (size_t i = 0; i < node.children.size(); ++i)
+    {
+        auto child = node.children[i];
+        if (const ASTExpressionList * exp_list = typeid_cast<const ASTExpressionList *>(&*child))
+            replaceConstFunction(*exp_list, context, all_columns);
+
+        if (const ASTFunction * function = typeid_cast<const ASTFunction *>(&*child))
+        {
+            auto result_block = getBlockWithConstants(function->clone(), context, all_columns);
```

Comment:
`getBlockWithConstants` doesn't change its first argument. What purpose of `clone`?